src/Microsoft.Fx.Portability/IAssemblyFile.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.Fx.Portability
+{
+    public interface IAssemblyFile
+    {
+        Stream OpenRead();
+
+        string Path { get; }
```

Comment:
Why path instead of full name? It seems like a name would be a more universally available. I think path is needed for a lot of our current MetadataReader-based code, but it we really want to support more general IAssembly-style inputs, we may need to allow for in-memory assemblies that report their own identities, etc.